src/meegsim/coupling.py

```diff
@@ -91,27 +100,61 @@ def ppc_von_mises(waveform, fs, phase_lag, *, kappa=1, m=1, n=1, fmin=None, fmax
     if not np.iscomplexobj(waveform):
         waveform = hilbert(waveform)
 
-    if fmin is None:
-        warnings.warn("fmin was None. Setting fmin to 8 Hz", UserWarning)
-        fmin = 8.
-    if fmax is None:
-        warnings.warn("fmax was None. Setting fmax to 12 Hz", UserWarning)
-        fmax = 12.
-
     waveform_amp = np.abs(waveform)
     waveform_angle = np.angle(waveform)
     n_samples = len(waveform)
 
     ph_distr = vonmises.rvs(kappa, loc=phase_lag, size=n_samples, random_state=random_state)
     tmp_waveform = np.real(waveform_amp * np.exp(1j * m / n * waveform_angle + 1j * ph_distr))
-    b, a = butter(N=2, Wn=np.array([m / n * fmin, m / n * fmax]) / fs * 2, btype='bandpass')
+    b, a = butter(N=2, Wn=np.array([m / n * fmin, m / n * fmax]) / sfreq * 2, btype='bandpass')
     tmp_waveform = filtfilt(b, a, tmp_waveform)
     waveform_coupled = waveform_amp * np.exp(1j * np.angle(hilbert(tmp_waveform)))
 
     return np.real(waveform_coupled)
 
 
+# This dictionary contains all built-in methods for simulating coupled time series
 COUPLING_FUNCTIONS = {
```

Comment:
Yeah, I'm also not very happy about this. Moving it does not seem possible because the coupling functions should be defined above, while the dispatcher should be defined below. I can extract it to a separate file, this should improve the structure.